core/math/transform_2d.cpp

```diff
@@ -251,7 +251,7 @@ Transform2D Transform2D::interpolate_with(const Transform2D &p_transform, real_t
 
 	real_t dot = v1.dot(v2);
 
-	dot = (dot < -1.0) ? -1.0 : ((dot > 1.0) ? 1.0 : dot); //clamp dot to [-1,1]
+	dot = CLAMP(dot, -1, 1);
```

Comment:
```suggestion
	dot = CLAMP(dot, -1.0, 1.0);
```

Some compilers are sensitive about those things, let's be explicit and avoid later warnings about type coercion.